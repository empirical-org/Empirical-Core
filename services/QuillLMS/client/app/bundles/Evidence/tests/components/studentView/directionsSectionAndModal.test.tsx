import * as React from 'react'
import { mount, } from 'enzyme'

import DirectionsSection from '../../../components/studentView/directionsSection'

const starterProps = {
  className: '',
  passage: { highlight_prompt: ''},
  inReflection: false,
  activeStep: 1
}

const inReflectionProps = { ...starterProps, inReflection: true }
const showReadTheDirectionsButtonProps = { ...starterProps, showReadTheDirectionsButton: true, }
const activeStepGreaterThanOneProps = { ...starterProps, activeStep: 2 }

describe('DirectionsSection component', () => {
  describe('when the student is on the read passage step', () => {
    it('renders', () => {
      const wrapper = mount(<DirectionsSection {...starterProps} />)
      expect(wrapper).toMatchSnapshot()
    })
  })

  describe('when the student is in reflection', () => {
    it('renders on desktop', () => {
      const wrapper = mount(<DirectionsSection {...inReflectionProps} />)
      expect(wrapper).toMatchSnapshot()
    })
  })

  describe('when the student should see the modal', () => {
    it('renders on desktop', () => {
      const wrapper = mount(<DirectionsSection {...showReadTheDirectionsButtonProps} />)
      expect(wrapper).toMatchSnapshot()
    })
  })

  describe('when the student is past the read passage step', () => {
    it('renders on desktop', () => {
      const wrapper = mount(<DirectionsSection {...activeStepGreaterThanOneProps} />)
      expect(wrapper).toMatchSnapshot()
    })
  })

})
