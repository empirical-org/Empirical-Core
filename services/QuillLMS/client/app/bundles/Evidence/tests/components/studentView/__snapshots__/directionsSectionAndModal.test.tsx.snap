// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DirectionsSection component when the student is in reflection renders on desktop 1`] = `
<DirectionsSection
  activeStep={1}
  className=""
  inReflection={true}
  passage={
    Object {
      "highlight_prompt": "",
    }
  }
>
  <div
    className="no-focus-outline "
    tabIndex={-1}
  >
    <section
      className="reflection-section"
    >
      <h3>
        Directions
      </h3>
      <p>
        Great! Now take a moment to reflect on the sentences you highlighted. The ideas you highlighted may be helpful as you complete the writing prompts in the next section.
      </p>
    </section>
  </div>
</DirectionsSection>
`;

exports[`DirectionsSection component when the student is on the read passage step renders 1`] = `
<DirectionsSection
  activeStep={1}
  className=""
  inReflection={false}
  passage={
    Object {
      "highlight_prompt": "",
    }
  }
>
  <div
    className="no-focus-outline "
    tabIndex={-1}
  >
    <section
      className="directions-section"
    >
      <h3>
        Directions
      </h3>
      <ul>
        <li>
          First, read the highlighting task below.
        </li>
        <li>
          Then, read the text and highlight sentences to complete the task.
        </li>
        <li
          className="sr-only"
        >
          Screenreader users, please use your tab keys to navigate through the passage. Highlightable sentences are toggle-able buttons. Use the Enter key on a focused sentence to add it to your highlights or to remove it once it has already been highlighted.
        </li>
      </ul>
    </section>
    <section
      className="task-section"
    >
      <h3>
        Task
      </h3>
      <p>
        As you read, highlight two sentences 
        <u />
      </p>
    </section>
  </div>
</DirectionsSection>
`;

exports[`DirectionsSection component when the student is past the read passage step renders on desktop 1`] = `
<DirectionsSection
  activeStep={2}
  className=""
  inReflection={false}
  passage={
    Object {
      "highlight_prompt": "",
    }
  }
>
  <section
    className="directions-section"
  >
    <h3>
      Directions
    </h3>
    <ul>
      <li>
        Use information 
        <b>
          from the text
        </b>
         to finish the sentence.
      </li>
      <li>
        Put the information in your own words.
      </li>
    </ul>
  </section>
</DirectionsSection>
`;

exports[`DirectionsSection component when the student should see the modal renders on desktop 1`] = `
<DirectionsSection
  activeStep={1}
  className=""
  inReflection={false}
  passage={
    Object {
      "highlight_prompt": "",
    }
  }
  showReadTheDirectionsButton={true}
>
  <div
    className="no-focus-outline "
    tabIndex={-1}
  >
    <section
      className="directions-section"
    >
      <h3>
        Directions
      </h3>
      <ul>
        <li>
          First, read the highlighting task below.
        </li>
        <li>
          Then, read the text and highlight sentences to complete the task.
        </li>
        <li
          className="sr-only"
        >
          Screenreader users, please use your tab keys to navigate through the passage. Highlightable sentences are toggle-able buttons. Use the Enter key on a focused sentence to add it to your highlights or to remove it once it has already been highlighted.
        </li>
      </ul>
    </section>
    <section
      className="task-section"
    >
      <h3>
        Task
      </h3>
      <p>
        As you read, highlight two sentences 
        <u />
      </p>
    </section>
  </div>
</DirectionsSection>
`;
